src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -73,8 +115,65 @@ public override void OnEnd(Activity activity)
 
             if (snapshot != pointers)
             {
+                var exceptionPointers = Marshal.PtrToStructure<EXCEPTION_POINTERS>(pointers);
+                var exceptionRecord = Marshal.PtrToStructure<EXCEPTION_RECORD>(exceptionPointers.ExceptionRecord);
+                activity.SetTag("exceptionRecord.ExceptionCode", exceptionRecord.ExceptionCode);
+                activity.SetTag("exceptionRecord.ExceptionFlags", exceptionRecord.ExceptionFlags);
+
+                switch (exceptionRecord.ExceptionCode)
+                {
+                case EXCEPTION_CODE.EXCEPTION_COMPLUS:
+                    int cntParameters = (int)exceptionRecord.NumberParameters;
+
+                    if (cntParameters <= 0 || cntParameters > 15)
+                    {
+                        break;
+                    }
+
+                    activity.SetTag("exceptionRecord.NumberParameters", exceptionRecord.NumberParameters);
+                    activity.SetTag("exceptionRecord.ExceptionInformation", exceptionRecord.ExceptionInformation);
```

Comment:
The actual information in the `ExceptionInformation` array is depending on the implementation:
* https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/coreclr/inc/ex.h#L38
* https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/coreclr/inc/ex.h#L49

Reply:
`ExceptionInformation[4]` has different meaning depending on the [condition compilation](https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/coreclr/utilcode/ex.cpp#L1323)

Reply:
`!sos.pe` might give some clue, worst case is that the logic is duplicated in `mscordacwks.dll`...